Nodejs/Product/Nodejs/Project/NodejsProjectNode.cs

```diff
@@ -202,7 +201,11 @@ public NodejsProjectNode(NodejsProjectPackage package)
 
         public bool IsTypeScriptProject {
             get {
-                return string.Equals(GetProjectProperty(NodejsConstants.EnableTypeScript), "true", StringComparison.OrdinalIgnoreCase);
+                var task = ProjectMgr.Site.GetUIThread().InvokeAsync(() => {
+                    return string.Equals(GetProjectProperty(NodejsConstants.EnableTypeScript), "true", StringComparison.OrdinalIgnoreCase);
+                });
+                task.Wait();
+                return task.Result;
```

Comment:
Yes, this must be called on the UI thread for some reason, which is why I created the separate variable to track this for the initial dts work. This latest change tried to reuse this variable so that they could not get out of sync.

I'm not sure the value can be cached though, since that will change the behavior of our existing code. What I can do though is move the task logic to the call sites of the property. That way, only that caller has the additional overhead.